commons/src/main/java/io/hyscale/commons/commands/CommandExecutor.java

```diff
@@ -15,100 +15,173 @@
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import io.hyscale.commons.config.SetupConfig;
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.exception.CommonErrorCode;
+import io.hyscale.commons.exception.HyscaleException;
 import io.hyscale.commons.models.CommandResult;
+import io.hyscale.commons.utils.HyscaleFilesUtil;
+import io.hyscale.commons.utils.ThreadPoolUtil;
 
+/**
+ * Class to execute commands using java process
+ * Use case:
+ * To execute command and check if it was successful
+ * To execute command and get output in String
+ * To execute command and copy output to a file
+ * To execute command in a particular directory
+ * 
+ * <p>
+ * In case file is not provided, an async thread is started
+ * to read copy command output to string writer
+ * This is done to ensure output buffer is not exhausted
+ * This thread runs until the process is alive
+ * String writer is used to get command output when required
+ * </p>
+ */
 @Component
 public class CommandExecutor {
 
-    private static final Logger logger = LoggerFactory.getLogger(CommandExecutor.class);
+	private static final Logger logger = LoggerFactory.getLogger(CommandExecutor.class);
+
+	@Autowired
+	private HyscaleFilesUtil filesUtil;
+
+	/**
+	 * @param command to be executed
+	 * @return {@link CommandResult} which contains command output and exit code
+	 */
+	public CommandResult executeAndGetResults(String command) {
+		CommandResult commandResult = null;
+		try {
+			commandResult = getCommandResult(command, null, null);
+		} catch (IOException | InterruptedException | HyscaleException e) {
+			HyscaleException ex = new HyscaleException(e, CommonErrorCode.FAILED_TO_EXECUTE_COMMAND, command);
+			logger.error("Failed while executing command, error {}", ex.toString());
+		}
+		return commandResult;
+	}
 
-    public CommandResult executeAndGetResults(String command) {
-        CommandResult commandResult = null;
-        int exitCode = 1;
-        try {
-            Process process = executeProcess(command, null, null);
-            exitCode = process.waitFor();
-            commandResult = new CommandResult();
-            commandResult
-                    .setCommandOutput(getResult(exitCode == 0 ? process.getInputStream() : process.getErrorStream()));
-            commandResult.setExitCode(exitCode);
-        } catch (IOException | InterruptedException e) {
-            HyscaleException ex = new HyscaleException(e, CommonErrorCode.FAILED_TO_EXECUTE_COMMAND, command);
-            logger.error("Failed while executing command, error {}", ex.toString());
-        }
-        return commandResult;
-    }
+	/**
+	 * @param command to be executed
+	 * @return whether command was successful or not
+	 */
+	public boolean execute(String command) {
+		CommandResult commandResult = null;
+		try {
+			commandResult = getCommandResult(command, null, null);
```

Comment:
Can we rename this method to some doExecute ?

Reply:
To keep consistency will update the names of other methods also.

Reply:
I meant the method getCommandResult()